### PR TITLE
Fix misleading logs in collator protocol.

### PR DESCRIPTION
**Pull request by eskimor**

### CODE REVIEW

File: node/network/collator-protocol/src/validator_side/mod.rs [hunk 1]
```diff
@@ -1210,10 +1210,11 @@ where
 			}
 			res = state.collation_fetch_timeouts.select_next_some() => {
 				let (collator_id, relay_parent) = res;
-				tracing::debug!(
+				tracing::trace!(
 					target: LOG_TARGET,
 					?relay_parent,
-					"Fetch for collation took too long, starting parallel download for next collator as well."
+					?collator_id,
+					"Timeout hit - already seconded?"
```
Comment (ordian) [inline node/network/collator-protocol/src/validator_side/mod.rs:1217]: can we check if it actually was seconded?
```rust
state.collations_per_relay_parent.get(&relay_parent).map_or(true, |s| matches!(s.status, CollationStatus::Seconded)) 
```

Comment (ordian) [reply node/network/collator-protocol/src/validator_side/mod.rs:1217]: And if it was seconded, we'd still try to fetch another one?

Comment (eskimor) [reply node/network/collator-protocol/src/validator_side/mod.rs:1217]: Whether or not another fetch is triggered (candidate was not yet seconded), can be determined by the trace log I added [below](https://github.com/paritytech/polkadot/pull/4201/files#diff-d1fc3f5e0ea52723187d9db009773910fbb0e15fb43e6f6a8e19e73f150ac5f9R1276).

Comment (eskimor) [reply node/network/collator-protocol/src/validator_side/mod.rs:1217]: In any case, as discussed with Rob - this code is going to go away anyways. For now, I just quickly wanted to get rid of that misleading log, without wasting too much time on this.

Comment (ordian) [reply node/network/collator-protocol/src/validator_side/mod.rs:1217]: I think it'd still be good to have them `debug!`, they shouldn't be too spammy.

**Review by ordian — APPROVED** [review]

**Pull request merged**